### PR TITLE
Fix Lora Rebase

### PR DESCRIPTION
**Pull request by hlahkar**
Fixes Lora Related issues in vllm Rebase

### CODE REVIEW

File: vllm/lora/punica.py [hunk 1]
```diff
@@ -131,9 +132,9 @@ def convert_mapping(
     if long_lora_context:
         assert long_lora_offsets is not None
         indices_list.append(long_lora_offsets)
-    indices = torch.tensor(indices_list, dtype=torch.long, device="cuda")
+    indices = torch.tensor(indices_list, dtype=torch.long, device=get_device())
```
Comment (kzawora-intel) [inline vllm/lora/punica.py:135]:
```suggestion
    device = 'hpu' if current_platform.is_hpu() else 'cuda'
    indices = torch.tensor(indices_list, dtype=torch.long, device=device)
```
Then use `device` instead of `get_device()` in the remainder of the file. For getting `current_platform`, add `from vllm.platforms import current_platform` at the top of the file.

File: tests/lora/test_lora_hpu.py [hunk 2]
```diff
@@ -1,221 +0,0 @@
-import pytest
```
Comment (vivekgoe) [inline tests/lora/test_lora_hpu.py:1]: Why did we delete test_lora_hpu.py? I know test has been deleted in upstream but let's keep it in vllm-fork. Eventually we probably will move HPU related tests to HPU extension repo.

File: vllm/lora/models.py [hunk 3]
```diff
@@ -26,9 +26,12 @@
                              parse_fine_tuned_lora_name, replace_submodule)
 from vllm.model_executor.models.interfaces import SupportsLoRA
 from vllm.model_executor.models.utils import PPMissingLayer
-from vllm.utils import is_pin_memory_available
+from vllm.utils import is_pin_memory_available, get_device
```
Comment (vivekgoe) [inline vllm/lora/models.py:29]: Where are we using get_device? If not needed, please delete.

File: vllm/lora/layers.py [hunk 4]
```diff
@@ -309,22 +226,7 @@ def set_lora(
...
-        if current_platform.is_hpu():
-            embedding_len = self.indices_len[3]
-            # NOTE(vgoel): These asserts can be skipped when upstreaming.
-            # Can be removed from vllm-fork also once lora functionality
-            # on Gaudi stabilizes.
-            if current_platform.is_hpu():
-                emb_len = embedding_len
-                x_shape = x.shape
-                ind_shape = self.embeddings_indices[1].shape
-                assert embedding_len == x.shape[0] * x.shape[1], \
```
Comment (vivekgoe) [inline vllm/lora/layers.py:321]: Can we move these asserts to overloaded function from PunicaWrapper class?

**Review by vivekgoe — CHANGES_REQUESTED** [review]

File: vllm/lora/models.py (hunk 3 above)
Comment (hlahkar) [reply vllm/lora/models.py:29]: get_device; is used in models.py; it is already part of habana_main; however this import was not part of the original rebase; hence putting it back; else we will get runtime error.

File: vllm/lora/layers.py (hunk 4 above)
Comment (hlahkar) [reply vllm/lora/layers.py:321]: this assert cannot be put in the PunicaWrapper class; as self.punica_wrapper.embeddings_indices does not have a access to x

File: vllm/lora/punica.py (hunk 1 above)
Comment (hlahkar) [reply vllm/lora/punica.py:135]: We are using get_device; to localize this if check; as get_device is used in many places in lora/models.py, lora/punica.py. This way the code looks clean

File: tests/lora/test_lora_hpu.py (hunk 2 above)
Comment (hlahkar) [reply tests/lora/test_lora_hpu.py:1]: Updated the test case using PunicaWrapper

Comment (hlahkar) [reply tests/lora/test_lora_hpu.py:1]: Updated the test using PunicaWrapper

**Comment by vivekgoe** [comment]
@michalkuligowski I have changed label from Intel to Habana. Both myself & Himangshu are from Habana team :-)
